packages/react/src/components/login/LoginCallbackHandler.test.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+
+import { getDefaultOidcClientTestProps } from './testUtils/oidcClientTestUtil';
+import { LoginContextProvider } from './LoginContext';
+import { LoginCallbackHandler } from './LoginCallbackHandler';
+import { Beacon, ConnectedModule } from './beacon/beacon';
+import { OidcClient, OidcClientState, oidcClientStates } from './client';
+import { getLastMockCallArgs } from '../../utils/testHelpers';
+import { isValidUser } from './client/oidcClient';
+import { triggerForAllOidcClientSignals } from './client/signals';
+import { createUser } from './testUtils/userTestUtil';
+import { OidcClientError } from './client/oidcClientError';
+
+const loginProps = getDefaultOidcClientTestProps();
+const onError = jest.fn();
+const onSuccess = jest.fn();
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+afterEach(() => {
+  jest.runOnlyPendingTimers();
+  jest.useRealTimers();
+  jest.restoreAllMocks();
+  sessionStorage.clear();
+});
+
+describe('LoginCallbackHandler', () => {
+  const notificationText = 'Logging in...';
+  let beacon: Beacon;
+  const renderComponent = (state: OidcClientState, returnUser: boolean) => {
+    const helperModule: ConnectedModule = {
+      namespace: 'helper',
+      connect: (targetBeacon) => {
+        beacon = targetBeacon;
+        beacon.addListener(triggerForAllOidcClientSignals, (signal) => {
+          const validUser = createUser();
+          const oidcClient = signal.context as OidcClient;
+          jest.spyOn(oidcClient, 'getState').mockReturnValue(state);
+          jest.spyOn(oidcClient, 'getUser').mockReturnValue(validUser);
+          if (!returnUser) {
+            jest.spyOn(oidcClient, 'handleCallback').mockRejectedValue(new Error('failed'));
```

Comment:
A lot of tests will fail, when function names change. Second argument for jest.spyOn must be a string.

Reply:
> A lot of tests will fail, when function names change.

Yeah, I guess I'm not a fan of spying on functions in tests because they break easily when refactoring, but they might be necessary in testing this type of code. 👍

Reply:
A good thing is that jest.spyOn will show a linting error when trying to spy a function which does not exist. So if function name changes, linting error will be thrown.